### ISSUE TITLE
[Feature Suggestion] A single docker image with all current LTS JDKs.

### ISSUE THREAD

**Issue by nielsbasjes** [adoptium/containers#274]
To illustrate my context:

I have a Java/Maven based project where I have
- a library part for which Java 8 is sufficient.
- some UDFs (to plug my library into various systems) where depending on that system either Java 8, 11 or 17 is "the only way to go". Some have Java 17 class files in their dependencies, some to things in their code no longer allowed in Java 17.

So for this project (where I want to build an test the code for all of this in one go) I have all LTS JDKs installed on my machine: 8, 11 and 17 and I'm using Maven Toolchains to use the correct one.

The key problem I ran into: 
- CI/CD systems like Gitlab-ci do the build using a docker image.
- There does not seem to be a docker image with all current LTS JDKs in it combined to support a usecase like this.

**Comment by brunoborges** [comment]
It sounds to me that the feature suggestion here is very specific to a very narrow use case, very specific to your particular need. In general, developers will have a build matrix in their CICD, like how [GitHub Actions supports](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs).

I'd love to understand how your particular scenario **cannot** be done with GitHub Actions matrices.

**Comment by nielsbasjes** [comment]
So what the build matrices do is build the code multiple times and then test that code with the same jvm version for which it was just built.

I believe that a matrix for a library that is published only once is a fundamentally incorrect approach to validate if it will work.

I want to build the code once and then verify this single build (which is what is published) works together with all jvms. 

This creates the need for having a build environment with all jdks for which the software needs to be tested.

After I posted this suggestion the GitHub action to setup the build environment was extended to support this. Also for Gitlab I have been able to make a workable solution.

See this https://github.com/nielsbasjes/ToolChainsInCiBuilds

So with that solution available it is fine if you close this ticket with a "won't fix"
